src/core/lib/security/security_connector/local/local_security_connector.cc

```diff
@@ -143,9 +143,13 @@ void local_check_peer(tsi_peer peer, grpc_endpoint* ep,
   }
   if (peer.properties != nullptr) gpr_free(peer.properties);
   peer.properties = new_properties;
-  // TODO(yihuazhang): Set security level of local TCP to TSI_SECURITY_NONE.
-  const char* security_level =
-      tsi_security_level_to_string(TSI_PRIVACY_AND_INTEGRITY);
+  // Set security level to NONE for TCP type, privacy&integrity otherwise.
+  const char* security_level;
+  if (type == LOCAL_TCP) {
```

Comment:
Right now this isn't a big problem because there are only 2 types, but to be more future-proof: can the default be "none" rather than "privacy & integrity"? Said differently, can we re-arrange this to:

```
if (type == UDS) {
  security_level = tsi_security_level_to_string(TSI_PRIVACY_AND_INTEGRITY);
} else {
  security_level = tsi_security_level_to_string(TSI_SECURITY_NONE);
}
```

Reply:
Ok, would it be better to use `switch with default` construction for it?

Reply:
Either is fine by me. Switch is probably more future-proof, good idea.